### PR TITLE
customizing room 

### PR DESCRIPTION
**Pull request by htak1**

### CODE REVIEW

File: src/edu/cmu/tartan/xml/XmlWriter.java [hunk 1]
```diff
@@ -129,23 +130,30 @@ public static void overWriteFile(String destFileName, String sourceFileName) thr
...
-        
-        is = new FileInputStream(source);
-        os = new FileOutputStream(dest);
-        byte[] buffer = new byte[1024];
-        int length;
-
-        while ((length = is.read(buffer)) > 0) {
-        	os.write(buffer, 0, length);
+        try (	InputStream is= new FileInputStream(source);
+        		OutputStream os= new FileOutputStream(dest);) {
```
Comment (tomcat1983) [inline src/edu/cmu/tartan/xml/XmlWriter.java]: Wrong indent
Absent white space between os and '='.

File: src/edu/cmu/tartan/xml/XmlWriter.java [hunk 2]
```diff
@@ -129,23 +130,30 @@ public static void overWriteFile(String destFileName, String sourceFileName) thr
...
+        	byte[] buffer = new byte[1024];
+        	int length;
+
+        	while ((length = is.read(buffer)) > 0) {
+        		os.write(buffer, 0, length);
+        	}
```
Comment (tomcat1983) [inline src/edu/cmu/tartan/xml/XmlWriter.java]: Wrong indent

File: src/edu/cmu/tartan/xml/XmlWriter.java [hunk 3]
```diff
@@ -129,23 +130,30 @@ public static void overWriteFile(String destFileName, String sourceFileName) thr
...
+        	close(is);
+        	close(os);
         }
+        catch (Exception e) {
+			gameInterface.severe("Exception occur when overWriteFile from " + sourceFileName + " to " + destFileName);
+        }
+	}
```
Comment (tomcat1983) [inline src/edu/cmu/tartan/xml/XmlWriter.java]: Wrong indent??

File: src/edu/cmu/tartan/xml/XmlWriter.java [hunk 4]
```diff
@@ -129,23 +130,30 @@ public static void overWriteFile(String destFileName, String sourceFileName) thr
...
-        OutputStream os = null;
-        
-        is = new FileInputStream(source);
-        os = new FileOutputStream(dest);
-        byte[] buffer = new byte[1024];
-        int length;
-
-        while ((length = is.read(buffer)) > 0) {
-        	os.write(buffer, 0, length);
+        try (	InputStream is= new FileInputStream(source);
```
Comment (tomcat1983) [inline src/edu/cmu/tartan/xml/XmlWriter.java]: Miss white space?
Unnecessary white space before InputStream.
Absent white space between is and '='.

File: src/edu/cmu/tartan/xml/XmlResponseUploadMap.java [hunk 5]
```diff
@@ -147,9 +146,11 @@ public XmlParseResult parsingRoomInfo(Document doc) {
 		for(i=0; i<roomCnt; i++) {
 
 			result = setAdjacentPath(nList, i);
-			if(result.equals(XmlParseResult.INVALID_DATA))
+			if( !result.equals(XmlParseResult.SUCCESS))
```
Comment (tomcat1983) [inline src/edu/cmu/tartan/xml/XmlResponseUploadMap.java]: Unnecessary white space

File: src/edu/cmu/tartan/xml/XmlResponseUploadMap.java [hunk 6]
```diff
@@ -258,25 +259,29 @@ else if(currentRoom instanceof RoomObscured) {
 			result = setRelationshipForRoomObscured((RoomObscured)currentRoom, nList, roomIndex);
 		}
 		
-		if(result.equals(XmlParseResult.INVALID_DATA)) {
-			return result; //don't need to proceed, it is already messed up 
+		if( result.equals(XmlParseResult.SUCCESS)) {
```
Comment (tomcat1983) [inline src/edu/cmu/tartan/xml/XmlResponseUploadMap.java]: Unnecessary white space

File: src/edu/cmu/tartan/xml/XmlResponseUploadMap.java [hunk 7]
```diff
@@ -258,25 +259,29 @@ else if(currentRoom instanceof RoomObscured) {
...
+			String hiddenItemStr= getAttributeValueAtNthTag("hidden_item", nList, roomIndex);
+			//for hiddenItem.  item_list="food:2" hidden_item="diamond"
+			if( hiddenItemStr != null )
```
Comment (tomcat1983) [inline src/edu/cmu/tartan/xml/XmlResponseUploadMap.java]: Unnecessary white space

File: src/edu/cmu/tartan/xml/XmlWriter.java (hunk 4 above)
Comment (htak1) [reply src/edu/cmu/tartan/xml/XmlWriter.java]: to match space with below statement

**Pull request merged**